### PR TITLE
Report pending and completion status to the associated VSTS commit

### PR DESCRIPTION
**Pull request by olivierdagenais**
This pull request adds a **Set pending status for VSTS commit or pull request** _build step_ as well as a **Set completion status for VSTS commit or pull request** _post-build action_, as shown below in my test job:

image

Manual testing
--------------

1. Triggered the `/vsts-hook/` with a specific commit, which queued my test job.
2. As soon as the Maven build started, I went to check on the commit in VSTS.  It had indeed received the "Pending" status.
3. As soon as the build completed successfully, I went to check on the commit in VSTS.  It had indeed received a "Success" status and was linking back to the Jenkins job.
4. I repeated the test, this time arranging for the build to fail.  VSTS indeed reported a "Failure" status, also linking back to the Jenkins job.

Mission accomplished!

### CODE REVIEW

File: src/main/java/hudson/plugins/tfs/VstsCompletedStatusPostBuildAction.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public static class DescriptorImpl extends BuildStepDescriptor<Publisher> {
+
+        @Override
+        public boolean isApplicable(final Class<? extends AbstractProject> jobType) {
+            return true;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Set completion status for VSTS commit or pull request";
```
Comment (DavidStaheli) [inline src/main/java/hudson/plugins/tfs/VstsCompletedStatusPostBuildAction.java]: Can we change "VSTS" to "Team Services" here?

File: src/main/java/hudson/plugins/tfs/VstsPendingStatusBuildStep.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
+
+        @Override
+        public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+            return true;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Set pending status for VSTS commit or pull request";
```
Comment (DavidStaheli) [inline src/main/java/hudson/plugins/tfs/VstsPendingStatusBuildStep.java]: Can we change "VSTS" to "Team Services" here?

**Comment by DavidStaheli** [comment]
Is there a way for these pre- and post-build actions to be automatic whenever TFS or Team Services is involved, so that administrators do not have to explicitly enable them per-job?

**Comment by olivierdagenais** [comment]
> Is there a way for these pre- and post-build actions to be automatic whenever TFS or Team Services is involved, so that administrators do not have to explicitly enable them per-job?

Yes, I think there’s a way and I’ve added it to the backlog.

File: src/main/java/hudson/plugins/tfs/VstsPendingStatusBuildStep.java (hunk 2 above)
Comment (olivierdagenais) [inline src/main/java/hudson/plugins/tfs/VstsPendingStatusBuildStep.java]: > Can we change "VSTS" to "Team Services" here?

If it's OK with you, I'll make a sweep after merging this and #92.

**Pull request merged**